File: 57/B.go
package main

import (
  "fmt"
  "math"
)

func main()  {
  var N,M int
  fmt.Scan(&N,&M)
  a := []int{}
  b := []int{}
  d := []int{}
  c := []int{}
  tmp,tmp2 := 0,0
  for i :=0; i < N ; i++ {
    fmt.Scan(&tmp,&tmp2)
    a = append(a,tmp)
    b = append(b,tmp2)
  }
  for i := 0; i < M; i++ {
    fmt.Scan(&tmp,&tmp2)
    c = append(c,tmp)
    d = append(d,tmp2)
  }
  var ans [N]int
  for i := 0; i < N; i++ {
    tmp := 10**9
    for j := 0; j < M; j++ {
      tmp = math.Min(tmp,math.Abs((a[i]-c[j]))+math.Abs(b[i]-d[j]))
    }
    a[i] = tmp
  }
  for i := 0; i < N; i++ {
    fmt.Println(a[i])
  }
}

File: 57/B.py
# encoding:utf-8
import copy
import random
import bisect #bisect_left　これで二部探索の大小検索が行える
import fractions #最小公倍数などはこっち
import math

mod = 10**9+7

N,M = map(int,input().split())
start = [0 for i in range(N)]
for i in range(N):
    start[i] = [int(i) for i in input().split()]

goal = [[] for i in range(M)]
for i in range(M):
    goal[i] = [int(i) for i in input().split()]

# print(start,goal)
ans = [0 for i in range(N)]
for i in range(N):
    tmp = mod
    for j in range(M):
        if tmp > abs(start[i][0]-goal[j][0])+abs(start[i][1]-goal[j][1]):
            ans[i] = j+1
            tmp = abs(start[i][0]-goal[j][0])+abs(start[i][1]-goal[j][1])
for i in range(N):
    print(ans[i])
